File: mcst/Statement.m
classdef Statement < Segment
    properties
        keyword char
        lvalue Expression
        rvalue Expression
        comment char
    end
    
    methods
        function self = Statement(keyword, lvalue, rvalue, comment)
            self.keyword = keyword;
            self.lvalue = lvalue;
            self.rvalue = rvalue;
            self.comment = comment;
        end
    end
end

File: test_m/py/nodes/Statement.py
from mruntime import *
from test_m.py.nodes.Segment import Segment
class Statement(Segment):
    def __init__(self, *nargin): # retval: self
        [keyword, modifier, lvalue, rvalue, comment] = nargin
        nargin = len(nargin)
        self.keyword = keyword
        self.modifier = modifier
        self.lvalue = lvalue
        self.rvalue = rvalue
        self.comment = comment
    @staticmethod
    def empty():
        return []
